File: customer-address-jpa/src/test/java/com/izertis/example/web/BaseWebTestClientTest.java
package com.izertis.example.web;

// import io.zenwave360.example.config.DockerComposeInitializer;
import com.izertis.example.config.DockerComposeInitializer;
import org.junit.jupiter.api.BeforeEach;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.web.reactive.server.WebTestClient;
import org.springframework.test.web.servlet.client.MockMvcWebTestClient;
import org.springframework.web.context.WebApplicationContext;

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK)
@ActiveProfiles("test")
@DockerComposeInitializer.EnableDockerCompose
@org.springframework.transaction.annotation.Transactional
public abstract class BaseWebTestClientTest {

    @Autowired
    protected WebApplicationContext context;
    protected WebTestClient webTestClient;

    @BeforeEach
    void setup() {
        this.webTestClient = MockMvcWebTestClient.bindToApplicationContext(this.context).build();
    }
}

File: kustomer-address-jpa/src/test/kotlin/com/izertis/example/web/BaseWebTestClientTest.kt
package com.izertis.example.web

// import com.izertis.examples.config.DockerComposeInitializer
import org.junit.jupiter.api.BeforeEach
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.ActiveProfiles
import org.springframework.test.web.reactive.server.WebTestClient
import org.springframework.test.web.servlet.client.MockMvcWebTestClient
import org.springframework.web.context.WebApplicationContext

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK)
@ActiveProfiles("test")
// @DockerComposeInitializer.EnableDockerCompose
@org.springframework.transaction.annotation.Transactional
abstract class BaseWebTestClientTest {

    @Autowired
    protected lateinit var context: WebApplicationContext
    protected lateinit var webTestClient: WebTestClient

    @BeforeEach
    fun setup() {
        this.webTestClient = MockMvcWebTestClient.bindToApplicationContext(this.context).build()
    }
}
